auth/clientcredentials/clientcredentials_test.go

```diff
@@ -0,0 +1,111 @@
+package clientcredentials
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"connectrpc.com/connect"
+	"github.com/hexops/autogold/v2"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2"
+
+	"github.com/sourcegraph/log/logtest"
+	sams "github.com/sourcegraph/sourcegraph-accounts-sdk-go"
+	clientsv1 "github.com/sourcegraph/sourcegraph-accounts-sdk-go/clients/v1"
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/clients/v1/clientsv1connect"
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/scopes"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type mockTokenIntrospector struct {
+	response *sams.IntrospectTokenResponse
+}
+
+func (m *mockTokenIntrospector) IntrospectToken(ctx context.Context, token string) (*sams.IntrospectTokenResponse, error) {
+	return m.response, nil
+}
+
+func TestInterceptor(t *testing.T) {
+	// All tests based on UsersService.GetUser()
+	for _, tc := range []struct {
+		name      string
+		token     *sams.IntrospectTokenResponse
+		wantError autogold.Value
+		wantLogs  autogold.Value
+	}{{
+		name: "inactive token",
+		token: &sams.IntrospectTokenResponse{
+			Active: false,
+		},
+		wantError: autogold.Expect("unauthenticated: invalid authorization header: invalid token type Bearer in Authorization header"),
+		wantLogs:  autogold.Expect([]string{}),
+	}, {
+		name: "insufficient scopes",
+		token: &sams.IntrospectTokenResponse{
+			Active: true,
+		},
+		wantError: autogold.Expect("unauthenticated: invalid authorization header: invalid token type Bearer in Authorization header"),
+		wantLogs:  autogold.Expect([]string{}),
+	}, {
+		name: "matches required scope",
+		token: &sams.IntrospectTokenResponse{
+			Active: true,
+			Scopes: scopes.Scopes{"profile"},
+		},
+		wantError: autogold.Expect("unauthenticated: invalid authorization header: invalid token type Bearer in Authorization header"), // should not error!
+		wantLogs:  autogold.Expect([]string{}),
+	}, {
+		name: "wrong scope",
+		token: &sams.IntrospectTokenResponse{
+			Active: true,
+			Scopes: scopes.Scopes{"not-a-scope"},
+		},
+		wantError: autogold.Expect("unauthenticated: invalid authorization header: invalid token type Bearer in Authorization header"),
```

Comment:
Are we sure we want errors to be asserted here be `invalid authorization header`? 😂 shouldn't them be like something `missing required scope: xxx`?

Reply:
oof good catch, I think I broke something

Reply:
I think https://github.com/sourcegraph/sourcegraph-accounts-sdk-go/pull/61/commits/e0d195ce3da572f65c7429c3f1392ab7602560db broke it (the equalfold change)

Reply:
condition was inverted hahaha

Reply:
updated!

Reply:
😂